repotracker/repotracker.go

```diff
@@ -589,3 +595,70 @@ func createVersionItems(v *version.Version, ref *model.ProjectRef, project *mode
 	}
 	return nil
 }
+
+func addBuildBreakSubscriptions(v *version.Version, projectRef *model.ProjectRef) error {
+	if !projectRef.NotifyOnBuildFailure {
+		return nil
+	}
+	subscriptionBase := event.Subscription{
+		Type:      event.ResourceTypeVersion,
+		Trigger:   "regression",
+		Selectors: trigger.MakeVersionSelectors(*v),
+	}
+	subscribers := []event.Subscriber{}
+
+	// if the commit author wants build break notifications, don't send to admins
+	if v.AuthorID != "" {
+		subscriber, _ := makeBuildBreakSubscriber(v.AuthorID)
```

Comment:
There is no need to make a build break subscriber for the user. Just grab their user settings document, and see if BuildBreakID.IsValid(). If it is, you're done and can return nil. You only need to makeBuildbreakSubscriber for admins.

Reply:
done